net/pdnsd/files/pdnsd.init

```diff
@@ -0,0 +1,32 @@
+#!/bin/sh /etc/rc.common
+# Copyright (C) 2019 OpenWrt.org
+
+USE_PROCD=1
+
+START=65
+STOP=10
+
+PROG=/usr/sbin/pdnsd
+CONFFILE=/etc/config/pdnsd.conf
+PIDFILE=/var/run/pdnsd.pid
+CACHEDIR=/tmp/pdnsd
+CACHEFILE=$CACHEDIR/pdnsd.cache
+
+start_service() {
+	gen_cache
+	procd_open_instance
+	procd_set_param command "$PROG" -c "$CONFFILE" -p "$PIDFILE"
+	procd_set_param limits nofile=4096
+	procd_set_param stdout 1
+	procd_set_param stderr 1
+	procd_set_param pidfile "$PIDFILE"
+	procd_close_instance
+}
+
+gen_cache()
+{
+	if [ ! -d "$CACHEDIR" ]; then
+		mkdir -p $CACHEDIR
+		dd if=/dev/zero of="$CACHEFILE" count=4 bs=1 2>/dev/null
```

Comment:
Why the cache needs to have 4 null bytes as initial content?  I am being curious.

If it's really needed, I may suggest using `bs=4 count=1` though in this case it's negligible.  I remembered using dd to copy large amount of data, bigger `bs` seems to be faster and less cpu overhead.

It's also okay if we just inline this function into `start_service`

Reply:
Fixed, remove dd and inline this function into start_service.